rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractor.java

```diff
@@ -0,0 +1,112 @@
+        Map<String, String> formWidgetKeyToTextMap = FormKeyTextExtractionUtil.getFormWidgetKeyToTextMap();
+        List<PatientHistoryEntry> patientHistoryEntries = new ArrayList<>();
+        EventClient eventClient = patientHistoryRepository.getEvent(baseEntityId, eventType, date);
+        if (eventClient != null) {
+            Event event = eventClient.getEvent();
+            for (Obs obs : event.getObs()) {
+                if (!shouldAddObs(obs, event)) {
+                    continue;
+                }
+                String text = formWidgetKeyToTextMap.get(obs.getFormSubmissionField());
+                if (StringUtils.isNotBlank(text)) {
+                    PatientHistoryEntry patientHistoryEntry = new PatientHistoryEntry(text,
+                            getValues(obs, formWidgetKeyToTextMap));
+                    patientHistoryEntries.add(patientHistoryEntry);
+                }
+            }
+        }
+        return patientHistoryEntries;
+    }
+
+    private boolean shouldAddObs(Obs obs, Event event) {
+        // do not process unique manual id if unique barcode id exists
+        boolean shouldIncludeManualIDField = isUniqueManualFormIDField(obs.getFormSubmissionField())
+                &&  event.findObs("", true, CovidConstants.FormFields.UNIQUE_ID) == null;
+
+        // do not process repeating group field, process it's corresponding _count field instead
+        boolean isNotRepeatingGroupField = !isUniqueManualFormIDField(obs.getFormSubmissionField())
+                && event.findObs("", true, obs.getFormSubmissionField() + "_count") == null;
+
+        return shouldIncludeManualIDField || isNotRepeatingGroupField;
+    }
+
+    private String getValues(Obs obs, Map<String, String> formWidgetKeyToTextMap) {
+        List<String> values = new ArrayList<>();
+        // for checkboxes, get keys from key-value map
+        Map<String, Object> keyValPairs = obs.getKeyValPairs();
+        List<Object> obsValues = Utils.isEmptyMap(keyValPairs) ? obs.getValues()
+                : new ArrayList<>(keyValPairs.keySet());
+        for (Object value : obsValues) {
+            values.add(getValue(value.toString(), formWidgetKeyToTextMap));
+        }
+        return StringUtils.join(values, ", ");
+    }
+
+    public JSONArray getJSONArr(String str) {
+        JSONArray jsonArray = null;
+        try {
+            jsonArray = new JSONArray(str);
+        } catch (JSONException ex) {
+            Timber.i("Invalid JSON array obs value.");
+        }
+        return jsonArray;
+    }
+
+    /**
+     * Takes care of the special case in native radio button and checkbox where the key should be used
+     *
+     * @param value
+     * @param formWidgetKeyToTextMap
+     * @return
+     */
+    private String getValue(String value, Map<String, String> formWidgetKeyToTextMap) {
+        // if value is a json string, return the length of json array (occurs in repeating groups)
```

Comment:
in this case do you ever see the contents of the array?

Reply:
Showing the contents of individual repeating group elements would require more processing.

I made changes to extract the number value associated with a repeating group instead.

Reply:
I don't understand, so is the answer then no, you do not see the contents of the array?

If no, what is normally inside these arrays, is it OK not to see it? That may also be best answered by someone else, unless it's obviously not interesting content, I don't know.

Reply:
https://github.com/onaio/rdt-standard/issues/319#issuecomment-661842540 this is what a repeating group looks like. 

You enter a number (`count`) and click the grey checkmark and it generates a layout (set of questions) repeated `count` number of times. Currently, we are only showing the `count` which makes sense since it's the answer to the initial question posed e.g. `Number of Covid-19 patients in household / compound? 3 (the count)`.

Responses recorded as part of each generated repeating group layout are not currently displayed.

A few reasons:

1. Extra logic is required to pull this information - repeating group data is not saved to one field so some collation would be required to match all the fields and reference them back to the parent repeating group.
2. I don't know if this is the correct place to show this information, it's dynamic and can grow large, possibly polluting this view.
3. And in reference to what you raised, it's not immediately clear that this is a necessary feature given the unknown LOE.

Reply:
@shivenbhatt1 can you weigh in on whether it's important to show this